pkg/cmd/cli/cmd/importimage.go

```diff
@@ -249,151 +249,134 @@ func (o *ImportImageOptions) waitForImport(resourceVersion string) (*imageapi.Im
 }
 
 func (o *ImportImageOptions) createImageImport() (*imageapi.ImageStream, *imageapi.ImageStreamImport, error) {
+	var isi *imageapi.ImageStreamImport
 	stream, err := o.isClient.Get(o.Name)
-	from := o.From
-	tag := o.Tag
-
+	// no stream, try creating one
 	if err != nil {
 		if !errors.IsNotFound(err) {
 			return nil, nil, err
 		}
 		if !o.Confirm {
```

Comment:
"I would agree, except 100% of new users would be annoyed by this, and only
a very small percentage of existing users would be helped by this"

On Tue, Jun 7, 2016 at 9:53 PM, Clayton Coleman <ccoleman@redhat.com> wrote:

> I would agree, except 100% of new users hit this, and only a very small
> percentage of existing users will ever hit this.  I don't have to run
> "--confirm" to run new-app, even though I might have made a typo.  I think
> --confirm on create new IS is overkill.
>
> On Tue, Jun 7, 2016 at 3:58 PM, Maciej Szulik <notifications@github.com>
> wrote:
>
>> In pkg/cmd/cli/cmd/importimage.go
>> <https://github.com/openshift/origin/pull/9163#discussion_r66141158>:
>>
>> >  	stream, err := o.isClient.Get(o.Name)
>> > -	from := o.From
>> > -	tag := o.Tag
>> > -
>> > +	// no stream, try creating one
>> >  	if err != nil {
>> >  		if !errors.IsNotFound(err) {
>> >  			return nil, nil, err
>> >  		}
>> >  		if !o.Confirm {
>>
>> Not quite sure, I'd prefer to leave it as is. I feel more comfortable
>> with warning users than forcing them to delete afterwards. They'll end up
>> wondering what happened and that's not desirable imho.
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/openshift/origin/pull/9163/files/6fa0d7e2d8abdfecbc63aa80f8ae37cb73e50bce..91256af4ae392aac584869ef57e3f9ea83f55045#r66141158>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe/ABG_p5uv2AZBcMg_nSj0Of41fJgJAskZks5qJc1YgaJpZM4ItsZW>
>> .
>>
>
>